File: clojure/grade-school/src/grade_school.clj
(ns grade-school)

(defn grade [school grade] 
  (get school grade []))


(defn add [school name grade] 
  (update school grade (fn [old] (if (nil? old) [name] (conj old name)))))

(defn sorted [school]  
  (into (sorted-map) (for [[k v] school] {k (sort v)})))

File: scala/grade-school/src/main/scala/GradeSchool.scala
import scala.collection.mutable.{Map => MMap}
class School {
  type DB = Map[Int, Seq[String]]
  val _db : DB = MMap()
  
  def add(name: String, g: Int) = 
    if (_db.contains(g)) {
      _db(g) = name :: _db(g)
    }else{
      _db(g) = Seq(name)
    }

  def db: DB = _db

  def grade(g: Int): Seq[String] = _db(g)

  def sorted: DB = ???
}

